frontend/src/modules/Folders/views/FolderView.js

```diff
@@ -51,13 +52,24 @@ function FolderPageHeader(props) {
 
   const goToS3Console = async () => {
     setIsLoadingUI(true);
-    const response = await client.query(
-      getDatasetAssumeRoleUrl(folder.dataset.datasetUri)
-    );
-    if (!response.errors) {
-      window.open(response.data.getDatasetAssumeRoleUrl, '_blank');
+    if (isAdmin) {
+      const response = await client.query(
+        getDatasetAssumeRoleUrl(folder.dataset.datasetUri)
+      );
+      if (!response.errors) {
+        window.open(response.data.getDatasetAssumeRoleUrl, '_blank');
+      } else {
+        dispatch({ type: SET_ERROR, error: response.errors[0].message });
+      }
     } else {
-      dispatch({ type: SET_ERROR, error: response.errors[0].message });
+      const response = await client.query(
+        getDatasetSharedAssumeRoleUrl(folder.dataset.datasetUri)
+      );
+      if (!response.errors) {
+        window.open(response.data.getDatasetSharedAssumeRoleUrl, '_blank');
+      } else {
+        dispatch({ type: SET_ERROR, error: response.errors[0].message });
+      }
```

Comment:
nit: I'd DRY this part with a small lambda

Reply:
what do you mean?

Reply:
it's the same code in if and else branch, with some different parameters, right? DRY means Don't Repeat Yourself.